### ISSUE TITLE
Image in Tooltip

### ISSUE THREAD

**Issue by hn1** [highcharts/highcharts#14056]
If we set useHTML: true for the tooltip and include an image, the image displays fine, but all the series data falls outside of the tooltip until you manually(not with js) resize the browser window.

 ```
tooltip: {
             useHTML: true,
             headerFormat: '<img src="/images/sample.jpg" width="150"/><br />► Listing {point.x}<br />',
             pointFormat: '<span style="color:{series.color}">► {series.name} : </span> {point.y:, .0f} <br />', 
             split: false,
             shared: true
          },

```

**Comment by pawelfus** [comment]
Hi @hn1 

Thanks for reporting the issue. This is caused by missing `height` option in `img` tag.

**Comment by hn1** [comment]
Hi
We have the height specified as :

headerFormat: '<img src="/images/man.jpg" style="height:100px" /><br />►
Listing {point.x}<br />',

**Comment by hn1** [comment]
This is a very interesting way of handling bug reports. Just Close it?

**Comment by hn1** [comment]
And this is it.
![hc](https://user-images.githubusercontent.com/8512745/90429007-cf066e00-e0bc-11ea-87b5-58238a5a3854.jpg)

**Comment by pawelfus** [comment]
Hi @hn1 

Thanks for the screenshot, but please, follow our template to report a bug. We can not debug from image or just a few lines of code. You should include all the necessary info:

![Screen Shot 2020-08-18 at 09 48 53](https://user-images.githubusercontent.com/1453926/90485629-11b95c00-e138-11ea-8b30-d58d83f7056a.png)

**Comment by hn1** [comment]
All you have to do is just try my code and you will see. No need for all
this protracted stuff. And if you can't be bothered - oh well.

**Comment by pawelfus** [comment]
@hn1 
I'm not bothered, I'm sorry if you felt that way!

It's just I don't know what exactly should I test. Setting height in your first post resolves the issue. Demo for you: https://jsfiddle.net/BlackLabel/repkLodf/

So all additional info (product Highcharts/Highstock/etc, product version and browser details) is important. Live demo (e.g. jsfiddle) is also very important, because allows us to debug your case, not only guess what is wrong.

Just to be on the same page, our bug-report template was create to speed up the process of helping each other. Deciding to not use it only protracts the process.

**Comment by hn1** [comment]
Many thanks for your reply.
Please take a look at this link: https://jsfiddle.net/BlackLabel/1xshj85w/
Thanks

**Comment by hn1** [comment]
Sorry that link has just the width.
Please remove the width and just enter the height and you will see the
issue.
Thanks

**Comment by sebastianbochan** [comment]
Hi @hn1,
The problem is caused by loading time of the image (and size). Image is initialized later than tooltip's calculation. As a result height is incorrect.

You can solve the problem by wrapping the image into a div with declaration of `height` or `min-height` params.

Demo:
- https://jsfiddle.net/BlackLabel/hrz6xLap/

**Comment by hn1** [comment]
WOW. You are a genius. Who else on earth would know that?

One last favor, is there any way to get the index of an xAxis Label from
the tooltip so that I can get the right image from data using that index
for each label.

P.S. I have tried other forums such as SO etc. but no joy.
Many thanks.

**Comment by sebastianbochan** [comment]
If you would like to add a condition, you should use the [tooltip.formatter](https://api.highcharts.com/highcharts/tooltip.formatter) instead of header/pointFormat.

I advise you to add conditions od `point.x` to manipulate the image.

Simple example:
- https://jsfiddle.net/BlackLabel/0ka5fpus/

For the further support questions, please use one of our channel like: stackoverflow / forum or email.

**Comment by hn1** [comment]
Many thanks.

So is it not possible to do it the way I wanted by using its index which
would be much better?
Regards

**Comment by sebastianbochan** [comment]
The `point.x` is index of point on xAxis. 

Do you min that image should switch when tick is changed?

**Comment by hn1** [comment]
Hi,
When I use poin.x, it is the Value of the xAxis and not the index.
Imagine having 100 points and 100 "if" statements.
With the index it is only one line of code. But I just can't get the index
of the label on the xAxis.
As I mentioned, point.x returns the Value and not the index.

**Comment by sebastianbochan** [comment]
I think that the best way is modify names of images.

See the PoC of it:
https://jsfiddle.net/BlackLabel/hnfe0cr1/

As a result you have references to images like: 
- 1800x1200_dog_cool_summer_other1.png
- 1800x1200_dog_cool_summer_other2.png
etc.

Does it make sense for you?

**Comment by sebastianbochan** [comment]
Hi @hn1,
Please elaborate what do you mean by index of xAxis label. 

Possibilities:
- index like [0,1,2] - first tick, second tick, third tick
- value of tick (which is displayed as label)

**Comment by hn1** [comment]
Hi
Yes we need the index of the xAxis label:
xAxis Index : 0, 1, 2, 3
xAxis Value : Apples, Oranges, Bananas

Many thanks.

**Comment by hn1** [comment]
Hi
Any news on this question so that we can create our charts.
Would appreciate a response.

**Comment by pawelfus** [comment]
Hi @hn1 

Let me explain how categories and values in Highcharts work, and why using label index is a bad idea.

In general, there's not direct relation between xAxis label and tooltip. Tooltip is rendered for point(s), not xAxis labels. When you use tooltip, you can use `point.x` as Sebastian said. That's the x-value on xAxis. If you are using categories on the chart, then it's an index of the category. Not an index of label. Given the fact, that you can create points between xAxis labels (e.g. datetime axis is a common use-case), relation label <-> point is unlikely to be used. It's rather x-value <-> point relation.

Now why indexing labels is a bad idea: no all labels are always visible. Basic example: https://jsfiddle.net/BlackLabel/kcxrLt1s/1/ - you may have so many points that some of the category labels are hidden. Using label index, you would refer to a wrong category. Imagine user viewing your chart on a mobile device, or just a small browser tab with many points. We don't provide label index because that might cause frustration in developers that indexes sometimes are changing (e.g. resizing browser).

In short, I'm pretty sure `point.x` is what you are looking for (as you want to generate images according to xAxis labels). Unfortunately, you didn't provide any demo which I could improve and show it to you, so I hope just the description will be enough.

Kind Regards,
Paweł Fus

**Comment by hn1** [comment]
Many thanks for your detailed response.
I will try and if I fail I will provide you with a sample.
Many thanks again.
